proto/desmos/posts/v1/models.proto

```diff
@@ -0,0 +1,258 @@
+}
+
+// PostReference contains the details of a post reference
+// TODO: Move this inside the Post object
+message PostReference {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  // Type of reference
+  Type type = 1;
+
+  // Id of the referenced post
+  uint64 post_id = 2 [ (gogoproto.customname) = "PostID" ];
+
+  // Type represents the different types of references
+  enum Type {
+    option (gogoproto.goproto_enum_prefix) = false;
+
+    // No reference specified
+    TYPE_UNSPECIFIED = 0;
+
+    // This post is a reply to the referenced post
+    TYPE_REPLIED_TO = 1;
+
+    // This post is a quote to the referenced post
+    TYPE_QUOTED = 2;
+
+    // This post is a report of the referenced post
+    TYPE_REPOSTED = 3;
+  }
+}
+
+// Contains the details of entities parsed out of the post text
+message Entities {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  repeated Tag hashtags = 1 [ (gogoproto.nullable) = false ];
+  repeated Tag mentions = 2 [ (gogoproto.nullable) = false ];
+  repeated Url urls = 3 [ (gogoproto.nullable) = false ];
+}
+
+// ReplySetting contains the possible reply settings that a post can have
+enum ReplySetting {
+  option (gogoproto.goproto_enum_prefix) = false;
+
+  // No reply setting specified
+  REPLY_SETTING_UNSPECIFIED = 0;
+
+  // Everyone will be able to reply to this post
+  REPLY_SETTING_EVERYONE = 1;
+
+  // Only followers of the author will be able to reply to this post
+  REPLY_SETTING_FOLLOWERS = 2;
+
+  // Only the author mutual followers will be able to reply to this post
+  REPLY_SETTING_MUTUAL = 3;
+
+  // Only people mentioned inside this post will be able to reply
+  REPLY_SETTING_MENTIONS = 4;
+}
+
+// Tag represents a generic tag
+message Tag {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  // Index of the character inside the text at which the tag starts
+  uint64 start = 1;
+
+  // Index of the character inside the text at which the tag ends
+  uint64 end = 2;
+
+  // Tag reference (user address, hashtag value, etc)
+  string tag = 3;
+}
+
+// Url contains the details of a generic URL
+message Url {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  // Index of the character inside the text at which the URL starts
+  uint64 start = 1;
+
+  // Index of the character inside the text at which the URL ends
+  uint64 end = 2;
+
+  // Value of the URL where the user should be redirected to
+  string url = 3;
+
+  // (optional) Display value of the URL
+  string display_url = 4;
+}
+
+// Attachment contains the data of a single post attachment
+message Attachment {
+  option (gogoproto.equal) = true;
+  option (gogoproto.goproto_stringer) = true;
+
+  // Id of the subspace inside which the post to which this attachment should be
+  // connected is is
+  uint64 subspace_id = 1 [ (gogoproto.customname) = "SubspaceID" ];
```

Comment:
Saving the subspace id inside the attachment itself results in a cleaner handling overall. A very nice example of how this could be useful to developers is if a DApp reads the data from two different subspaces (for any reason): it this case they might have multiple attachments with the same `post_id` and `id` values, unless we return the `subspace_id` inside the query response or somehow they manage this by themselves. This is way harder than simply storing a `uint64` value (just 8 bytes) inside the storage itself in my opinion

Reply:
Oh. Right. I didn't think it this way. Yes it's definitely better to handle this with the additional `subspace_id` rather than leaving developers handle it in many different ways. 
It would only bring chaos and makes thing messy for no reason.